<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react_perf/jsx-no-jsx-as-prop <Badge type="info" text="Perf" />

<div class="rule-meta">
</div>

### What it does

Prevent JSX that are local to the current method from being used as values of JSX props

### Example

```jsx
// Bad
<Item jsx={<SubItem />} />
<Item jsx={this.props.jsx || <SubItem />} />
<Item jsx={this.props.jsx ? this.props.jsx : <SubItem />} />

// Good
<Item callback={this.props.jsx} />
```
