<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react/button-has-type <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Enforces explicit `type` attribute for all the `button` HTML elements.

### Why is this bad?

The default value of `type` attribute for `button` HTML element is
`"submit"` which is often not the desired behavior and may lead to
unexpected page reloads.

### Example

```jsx
// Bad
<button />
<button type="foo" />

// Good
<button type="button" />
<button type="submit" />
```
