<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react/jsx-no-useless-fragment <Badge type="info" text="Pedantic" />

<div class="rule-meta">
</div>

### What it does

Disallow unnecessary fragments.

### Why is this bad?

Fragments are a useful tool when you need to group multiple children without adding a node to the DOM tree. However, sometimes you might end up with a fragment with a single child. When this child is an element, string, or expression, it's not necessary to use a fragment.

### Example

```jsx
// Bad
<>foo</>
<div><>foo</></div>

// Good
<>foo <div></div></>
<div>foo</div>
```
