<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-assign-module-variable <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

### Why is this bad?

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/a6b0100501fda75ec313146a992a9f5fce995518/crates/oxc_linter/src/rules/nextjs/no_assign_module_variable.rs)
