<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-before-interactive-script-outside-document <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent usage of `next/script`'s `beforeInteractive` strategy outside of `pages/_document.js`.

### Why is this bad?

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/e91c2878d8c49213790df4d192bb3136503aa08b/crates/oxc_linter/src/rules/nextjs/no_before_interactive_script_outside_document.rs)
