<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-css-tags <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

### Why is this bad?

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/45ef3eb2becc663e756a3da557df5a03fb40fd84/crates/oxc_linter/src/rules/nextjs/no_css_tags.rs)
