<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-document-import-in-page <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent importing `next/document` outside of `pages/_document.js`.

### Why is this bad?

### Example

```javascript

```
