<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-document-import-in-page <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent importing `next/document` outside of `pages/_document.js`.

### Why is this bad?

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/5190b7fb280dcd15c6de81717c073b06a161ced6/crates/oxc_linter/src/rules/nextjs/no_document_import_in_page.rs)
