<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-duplicate-head <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent duplicate usage of `<Head>` in `pages/\_document.js``.

### Why is this bad?

This can cause unexpected behavior in your application.

### Example

```jsx
import Document, { Html, Head, Main, NextScript } from "next/document";
class MyDocument extends Document {
  static async getInitialProps(ctx) {}
  render() {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
export default MyDocument;
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/55744fd9775a032ea83215dadc77a6bff3e2a87d/crates/oxc_linter/src/rules/nextjs/no_duplicate_head.rs)
