<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-head-element <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent usage of `<head>` element.

### Why is this bad?

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/25ddb3519289dfe2a51e53b9b0c355d8ad950bbc/crates/oxc_linter/src/rules/nextjs/no_head_element.rs)
