<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-head-element <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent usage of `<head>` element.

### Why is this bad?

### Example

```javascript

```
