<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-page-custom-font <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent page-only custom fonts.

### Why is this bad?

- The custom font you're adding was added to a page - this only adds the font to the specific page and not the entire application.
- The custom font you're adding was added to a separate component within pages/\_document.js - this disables automatic font optimization.

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/e91c2878d8c49213790df4d192bb3136503aa08b/crates/oxc_linter/src/rules/nextjs/no_page_custom_font.rs)
