<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-styled-jsx-in-document <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent usage of styled-jsx in pages/\_document.js.

### Why is this bad?

Custom CSS like styled-jsx is not allowed in a [Custom Document](https://nextjs.org/docs/pages/building-your-application/routing/custom-document).

### Example

```javascript

```
