<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-title-in-document-head <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

### Why is this bad?

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/55744fd9775a032ea83215dadc77a6bff3e2a87d/crates/oxc_linter/src/rules/nextjs/no_title_in_document_head.rs)
