<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-typos <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent common typos in Next.js's data fetching functions

### Why is this bad?

### Example

```javascript
export default function Page() {
  return <div></div>;
}
export async function getServurSideProps() {}
```
