<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-unwanted-polyfillio <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Prevent duplicate polyfills from Polyfill.io.

### Why is this bad?

You are using polyfills from Polyfill.io and including polyfills already shipped with Next.js. This unnecessarily increases page weight which can affect loading performance.

### Example

```javascript
<script src='https://polyfill.io/v3/polyfill.min.js?features=Array.prototype.copyWithin'></script>

<script src='https://polyfill.io/v3/polyfill.min.js?features=WeakSet%2CPromise%2CPromise.prototype.finally%2Ces2015%2Ces5%2Ces6'></script>
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/55744fd9775a032ea83215dadc77a6bff3e2a87d/crates/oxc_linter/src/rules/nextjs/no_unwanted_polyfillio.rs)
