<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsdoc/check-access <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Checks that `@access` tags use one of the following values:

- "package", "private", "protected", "public"

Also reports:

- Mixing of `@access` with `@public`, `@private`, `@protected`, or `@package` on the same doc block.
- Use of multiple instances of `@access` (or the `@public`, etc) on the same doc block.

### Why is this bad?

It is important to have a consistent way of specifying access levels.

### Example

```javascript
// Passing
/** @access private */

/** @private */

// Failing
/** @access private @public */

/** @access invalidlevel */
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/jsdoc/check_access.rs)
