<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsdoc/check-property-names <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Ensures that property names in JSDoc are not duplicated on the same block and that nested properties have defined roots.

### Why is this bad?

`@property` tags with the same name can be confusing and may indicate a mistake.

### Example

```javascript
// Passing
/**
 * @typedef {object} state
 * @property {number} foo
 */
/**
 * @typedef {object} state
 * @property {object} foo
 * @property {number} foo.bar
 */

// Failing
/**
 * @typedef {object} state
 * @property {number} foo
 * @property {string} foo
 */

/**
 * @typedef {object} state
 * @property {number} foo.bar
 */
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/jsdoc/check_property_names.rs)
