<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsdoc/check-tag-names <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Reports invalid block tag names.
Additionally checks for tag names that are redundant when using a type checker such as TypeScript.

### Why is this bad?

Using invalid tags can lead to confusion and make the documentation harder to read.

### Example

```javascript
// Passing
/** @param */

// Failing
/** @Param */
/** @foo */

/**
 * This is redundant when typed.
 * @type {string}
 */
```
