<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsdoc/require-property <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Requires that all `@typedef` and `@namespace` tags have `@property` tags
when their type is a plain `object`, `Object`, or `PlainObject`.

### Why is this bad?

Object type should have properties defined.

### Example

```javascript
// Passing
/**
 * @typedef {Object} SomeTypedef
 * @property {SomeType} propName Prop description
 */
/**
 * @typedef {object} Foo
 * @property someProp
 */

// Failing
/**
 * @typedef {Object} SomeTypedef
 */
/**
 * @namespace {Object} SomeNamesoace
 */
```
