<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsdoc/require-property-description <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Requires that all `@property` tags have descriptions.

### Why is this bad?

The description of a property should be documented.

### Example

```javascript
// Passing
/**
 * @typedef {SomeType} SomeTypedef
 * @property {number} foo Foo.
 */

// Failing
/**
 * @typedef {SomeType} SomeTypedef
 * @property {number} foo
 */
```
