<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# vitest/no-import-node-test <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">🛠️</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

This rule warns when `node:test` is imported (usually accidentally). With `--fix`, it will replace the import with `vitest`.

### Examples

```javascript
// invalid
import { test } from "node:test";
import { expect } from "vitest";

test("foo", () => {
  expect(1).toBe(1);
});
```

```javascript
// valid
import { test, expect } from "vitest";

test("foo", () => {
  expect(1).toBe(1);
});
```
