<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-magic-array-flat-depth <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Disallow magic numbers for `Array.prototype.flat` depth.

### Why is this bad?

Magic numbers are hard to understand and maintain. When calling `Array.prototype.flat`, it is usually called with `1` or infinity. If you are using a different number, it is better to add a comment explaining the depth.

### Example

```javascript
// Bad
array.flat(2);
array.flat(20);
//
// Good
array.flat(2 /* explanation */);
array.flat(1);
array.flat();
array.flat(Infinity);
```
