<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-negation-in-equality-check <Badge type="info" text="Pedantic" />

<div class="rule-meta">
</div>

### What it does

Disallow negated expressions on the left of (in)equality checks.

### Why is this bad?

A negated expression on the left of an (in)equality check is likely a mistake from trying to negate the whole condition.

### Example

```javascript
// Bad

if (!foo === bar) {
}

if (!foo !== bar) {
}

// Good

if (foo !== bar) {
}

if (!(foo === bar)) {
}
```
