<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-static-only-class <Badge type="info" text="Pedantic" />

<div class="rule-meta">
</div>

### What it does

Disallow classes that only have static members.

### Why is this bad?

A class with only static members could just be an object instead.

### Example

```javascript
// Bad
class A {
  static a() {}
}

// Good
class A {
  static a() {}

  constructor() {}
}
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/unicorn/no_static_only_class.rs)
