<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-unreadable-array-destructuring <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Disallow unreadable array destructuring

### Why is this bad?

Destructuring is very useful, but it can also make some code harder to read.
This rule prevents ignoring consecutive values when destructuring from an array.

### Example

Examples of **incorrect** code for this rule:

```javascript
const [, , foo] = parts;
```

Examples of **correct** code for this rule:

```javascript
const [foo] = parts;
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/a6b0100501fda75ec313146a992a9f5fce995518/crates/oxc_linter/src/rules/unicorn/no_unreadable_array_destructuring.rs)
