<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-useless-switch-case <Badge type="info" text="Pedantic" />

<div class="rule-meta">
</div>

### What it does

Disallows useless default cases in switch statements.

### Why is this bad?

An empty case before the last default case is useless.

### Example

```javascript
// bad
switch (foo) {
  case 1:
  default:
    handleDefaultCase();
    break;
}
// good:
switch (foo) {
  case 1:
  case 2:
    handleCase1And2();
    break;
}
```
