<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-unreadable-array-destructuring <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Disallow unreadable array destructuring

### Why is this bad?

Destructuring is very useful, but it can also make some code harder to read.
This rule prevents ignoring consecutive values when destructuring from an array.

### Example

```javascript
// Bad
const [, , foo] = parts;

// Good
const [foo] = parts;
```
