<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-reflect-apply <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

### Why is this bad?

Reflect.apply() is arguably less verbose and easier to understand.
In addition, when you accept arbitrary methods,
it's not safe to assume .apply() exists or is not overridden.

### Example

Examples of **incorrect** code for this rule:

```javascript
foo.apply(null, [42]);
```

Examples of **correct** code for this rule:

```javascript
Reflect.apply(foo, null);
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/25ddb3519289dfe2a51e53b9b0c355d8ad950bbc/crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs)
