<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-modern-dom-apis <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Enforces the use of:

- `childNode.replaceWith(newNode)` over `parentNode.replaceChild(newNode, oldNode)`
- `referenceNode.before(newNode)` over `parentNode.insertBefore(newNode, referenceNode)`
- `referenceNode.before('text')` over `referenceNode.insertAdjacentText('beforebegin', 'text')`
- `referenceNode.before(newNode)` over `referenceNode.insertAdjacentElement('beforebegin', newNode)`

### Why is this bad?

There are some advantages of using the newer DOM APIs, like:

- Traversing to the parent node is not necessary.
- Appending multiple nodes at once.
- Both `DOMString` and DOM node objects can be manipulated.

### Example

```javascript
// Bad
oldChildNode.replaceWith(newChildNode);

// Good
parentNode.replaceChild(newChildNode, oldChildNode);
```
