<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-reflect-apply <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

### Why is this bad?

Reflect.apply() is arguably less verbose and easier to understand.
In addition, when you accept arbitrary methods,
it's not safe to assume .apply() exists or is not overridden.

### Example

```javascript
// Bad
foo.apply(null, [42]);

// Good
Reflect.apply(foo, null);
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs)
