<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-logical-operator-over-ternary <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

This rule finds ternary expressions that can be simplified to a logical operator.

### Why is this bad?

Using a logical operator is shorter and simpler than a ternary expression.

### Example

```javascript
// Bad
const foo = bar ? bar : baz;
console.log(foo ? foo : bar);

// Good
const foo = bar || baz;
console.log(foo ?? bar);
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/unicorn/prefer_logical_operator_over_ternary.rs)
