<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-includes <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Prefer `includes()` over `indexOf()` when checking for existence or non-existence.

All built-ins have `.includes()` in addition to `.indexOf()`.

### Why is this bad?

The `.includes()` method is more readable and less error-prone than `.indexOf()`.

### Example

```javascript
// bad
if (str.indexOf("foo") !== -1) {
}

// good
if (str.includes("foo")) {
}
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/unicorn/prefer_includes.rs)
