<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/throw-new-error <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Require `new` when throwing an error.`

### Why is this bad?

While it's possible to create a new error without using the `new` keyword, it's better to be explicit.

### Examples

Examples of **incorrect** code for this rule:

```javascript
throw Error("ü¶Ñ");
throw TypeError("unicorn");
throw lib.TypeError("unicorn");
```

Examples of **correct** code for this rule:

```javascript
throw new Error("ü¶Ñ");
throw new TypeError("unicorn");
throw new lib.TypeError("unicorn");
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/a6b0100501fda75ec313146a992a9f5fce995518/crates/oxc_linter/src/rules/unicorn/throw_new_error.rs)
