<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# import/default <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

If a default import is requested, this rule will report if there is no default export in the imported module.

### Example

```javascript
// ./bar.js
export function bar() {
  return null;
}

// ./foo.js
import bar from "./bar"; // no default export found in ./bar
```
