<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# import/no-dynamic-require <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Forbid imports which use an expression for the module argument.

### Why is this bad?

Import statements which use an expression resolved at runtime makes it to find where the
import comes from and some static code analysis tools might not be able to resolve them.

### Examples

Examples of **incorrect** code for this rule:

```javascript
require(name);
require(`../${name}`);
```

Examples of **correct** code for this rule:

```javascript
require("../name");
require(`../name`);
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/import/no_dynamic_require.rs)
