<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# import/no-named-as-default <Badge type="info" text="Suspicious" />

<div class="rule-meta">
</div>

### What it does

Reports use of an exported name as the locally imported name of a default export.

### Example

```javascript
// foo.js
export default "foo";
export const bar = "baz";
```

Valid:

```javascript
import foo from "./foo.js";
```

Invalid:

```javascript
// using exported name 'bar' as identifier for default export.
import bar from "./foo.js";
```
