<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# oxc/number-arg-out-of-range <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Checks whether the radix or precision arguments of number-related functions exceeds the limit.

### Example

```javascript
var x = 42;
var s_radix_64 = x.toString(64);
var s = x.toString(1);
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/a6b0100501fda75ec313146a992a9f5fce995518/crates/oxc_linter/src/rules/oxc/number_arg_out_of_range.rs)
