<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/default-param-last <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Enforce default parameters to be last

### Why is this bad?

Putting default parameter at last allows function calls to omit optional tail arguments.

### Example

```javascript
// Correct: optional argument can be omitted
function createUser(id, isAdmin = false) {}
createUser("tabby");

// Incorrect: optional argument can **not** be omitted
function createUser(isAdmin = false, id) {}
createUser(undefined, "tabby");
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/2b14a6fb5403dfe5eff8dfade3baf7725b837052/crates/oxc_linter/src/rules/eslint/default_param_last.rs)
