<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-caller <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow the use of arguments.caller or arguments.callee

### Why is this bad?

The use of arguments.caller and arguments.callee make several code optimizations impossible.
They have been deprecated in future versions of JavaScript and their use is forbidden in ECMAScript 5 while in strict mode.

### Example

```javascript
function foo(n) {
  if (n <= 0) {
    return;
  }

  arguments.callee(n - 1);
}

[1, 2, 3, 4, 5].map(function (n) {
  return !(n > 1) ? 1 : arguments.callee(n - 1) * n;
});
```
