<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-const-assign <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow reassigning const variables

### Why is this bad?

We cannot modify variables that are declared using const keyword.
It will raise a runtime error.

### Example

```javascript
const a = 0;
a = 1;
```
