<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-dupe-keys <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow duplicate keys in object literals

### Why is this bad?

Multiple properties with the same key in object literals can cause unexpected behavior in your application.

### Example

```javascript
var foo = {
  bar: "baz",
  bar: "qux",
};
```
