<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-empty-function <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Disallows the usages of empty functions

### Why is this bad?

Empty functions can reduce readability because readers need to guess whether itâ€™s
intentional or not. So writing a clear comment for empty functions is a good practice.

### Example

Examples of **incorrect** code for this rule:

```javascript
function foo() {}

const bar = () => {};
```

Examples of **correct** code for this rule:

```javascript
function foo() {
  // do nothing
}

function foo() {
  return;
}
const add = (a, b) => a + b;
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/5190b7fb280dcd15c6de81717c073b06a161ced6/crates/oxc_linter/src/rules/eslint/no_empty_function.rs)
