<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-empty-function <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Disallows the usages of empty functions

### Why is this bad?

Empty functions can reduce readability because readers need to guess whether itâ€™s
intentional or not. So writing a clear comment for empty functions is a good practice.

### Example

```javascript
function foo() {}

const bar = () => {};
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/eslint/no_empty_function.rs)
