<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-global-assign <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">✅</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow modifications to read-only global variables.

### Why is this bad?

In almost all cases, you don’t want to assign a value to these global variables as doing so could result in losing access to important functionality.

### Example

```javascript
Object = null;
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/57b7ca8eae8a1d308119ac0e0a8e657e5e61b4bb/crates/oxc_linter/src/rules/eslint/no_global_assign.rs)
