<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-loss-of-precision <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow precision loss of number literal

### Why is this bad?

It can lead to unexpected results in certain situations
For example, when performing mathematical operations

### Example

```javascript
var x = 2e999;
```
