<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-new-native-nonconstructor <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow new operators with global non-constructor functions (Symbol, BigInt)

### Why is this bad?

Both new Symbol and new BigInt throw a type error because they are functions and not classes.
It is easy to make this mistake by assuming the uppercase letters indicate classes.

### Example

```javascript
// throws a TypeError
let foo = new Symbol("foo");

// throws a TypeError
let result = new BigInt(9007199254740991);
```
