<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-sparse-arrays <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">✅</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow sparse arrays.

### Why is this bad?

The confusion around sparse arrays is enough that it’s recommended to avoid using them unless you are certain that they are useful in your code.

### Example

```javascript
var items = [, ,];
var colors = ["red", , "blue"];
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/5190b7fb280dcd15c6de81717c073b06a161ced6/crates/oxc_linter/src/rules/eslint/no_sparse_arrays.rs)
