<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-unsafe-negation <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">‚úÖ</span> This rule is turned on by default.
</Alert>
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Disallow negating the left operand of relational operators

### Why is this bad?

Just as developers might type -a + b when they mean -(a + b) for the negative of a sum,
they might type !key in object by mistake when they almost certainly mean !(key in object)
to test that a key is not in an object. !obj instanceof Ctor is similar.

### Example

```javascript
if ((!key) in object) {
  //operator precedence makes it equivalent to (!key) in object
  //and type conversion makes it equivalent to (key ? "false" : "true") in object
}
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/25ddb3519289dfe2a51e53b9b0c355d8ad950bbc/crates/oxc_linter/src/rules/eslint/no_unsafe_negation.rs)
