<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# promise/valid-params <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Enforces the proper number of arguments are passed to Promise functions.

### Why is this bad?

Calling a Promise function with the incorrect number of arguments can lead to unexpected
behavior or hard to spot bugs.

### Example

```javascript
Promise.resolve(1, 2);
```
