<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jest/prefer-called-with <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Suggest using `toBeCalledWith()` or `toHaveBeenCalledWith()`

### Example

```javascript
// valid
expect(noArgsFunction).toBeCalledWith();
expect(roughArgsFunction).toBeCalledWith(expect.anything(), expect.any(Date));
expect(anyArgsFunction).toBeCalledTimes(1);
expect(uncalledFunction).not.toBeCalled();

// invalid
expect(someFunction).toBeCalled();
expect(someFunction).toHaveBeenCalled();
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/55744fd9775a032ea83215dadc77a6bff3e2a87d/crates/oxc_linter/src/rules/jest/prefer_called_with.rs)
