<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jest/valid-title <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">🛠️</span> An auto-fix is available for this rule for some violations.
</Alert>
</div>

### What it does

Checks that the titles of Jest blocks are valid.

Titles must be:

- not empty,
- strings,
- not prefixed with their block name,
- have no leading or trailing spaces.

Why is this bad?

Titles that are not valid can be misleading and make it harder to understand the purpose of the test.

### Example

Examples of **incorrect** code for this rule:

```javascript
describe("", () => {});
describe("foo", () => {
  it("", () => {});
});
it("", () => {});
test("", () => {});
xdescribe("", () => {});
xit("", () => {});
xtest("", () => {});
```

Examples of **correct** code for this rule:

```javascript
describe('foo', () => {});
it('bar', () => {});
test('baz', () => {});


## References
- [Rule Source](https://github.com/oxc-project/oxc/blob/55744fd9775a032ea83215dadc77a6bff3e2a87d/crates/oxc_linter/src/rules/jest/valid_title.rs)
```
