<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/aria-unsupported-elements <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Certain reserved DOM elements do not support ARIA roles, states and
properties. This is often because they are not visible, for example
`meta`, `html`, `script`, `style`. This rule enforces that these DOM
elements do not contain the `role` and/or `aria-*` props.

### Example

```jsx
// Good
	<meta charset="UTF-8" />

// Bad
<meta charset="UTF-8" aria-hidden="false" />
```
