<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/autocomplete-valid <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Enforces that an element's autocomplete attribute must be a valid value.

### Why is this bad?

Incorrectly using the autocomplete attribute may decrease the accessibility of the website for users.

### Example

```jsx
// Bad
<input autocomplete="invalid-value" />

// Good
<input autocomplete="name" />
```
