<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/mouse-events-have-key-events <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Enforce onmouseover/onmouseout are accompanied by onfocus/onblur.

### Why is this bad?

Coding for the keyboard is important for users with physical disabilities who cannot use a mouse,
AT compatibility, and screenreader users.

### Example

```jsx
// Good
<div onMouseOver={() => void 0} onFocus={() => void 0} />

// Bad
<div onMouseOver={() => void 0} />
```
