<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/no-autofocus <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Enforce that `autoFocus` prop is not used on elements. Autofocusing
elements can cause usability issues for sighted and non-sighted users,
alike.

### Rule Option

This rule takes one optional object argument of type object:

```json
{
  "rules": {
    "jsx-a11y/no-autofocus": [
      2,
      {
        "ignoreNonDOM": true
      }
    ]
  }
}
```

For the `ignoreNonDOM` option, this determines if developer created
components are checked.

### Example

Examples of **incorrect** code for this rule:

```jsx
<div autoFocus />
<div autoFocus="true" />
<div autoFocus="false" />
<div autoFocus={undefined} />
```

Examples of **correct** code for this rule:

```jsx
<div />
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/5190b7fb280dcd15c6de81717c073b06a161ced6/crates/oxc_linter/src/rules/jsx_a11y/no_autofocus.rs)
