<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/no-redundant-roles <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Enforces that the explicit `role` property is not the same as
implicit/default role property on element.

### Why is this bad?

Redundant roles can lead to confusion and verbosity in the codebase.

### Example

```jsx
// Bad
<nav role="navigation" />

// Good
<nav />
```
